kotlinpoet/src/main/java/com/squareup/kotlinpoet/TypeName.kt

```diff
@@ -383,7 +383,7 @@ class ClassName internal constructor(
     }
     // concat top level class name and nested names
     return buildString {
-      append(topLevelClassName())
+      append(topLevelClassName().canonicalName)
```

Comment:
Without the `canonicalName` it would escape the topLevelClassName. Since this is used in reflection, escaping isn't required and would throw an exception